### PR TITLE
Hi

### PR DESCRIPTION
**Pull request by antonbobkov**

### CODE REVIEW

File: erlang/compile_script [hunk 1]
```diff
@@ -0,0 +1 @@
+ erlc -o beam/ `find . -not -path './p2p/*' -name '*.erl'`
```
Comment (pdlla) [inline erlang/compile_script]: :o

File: erlang/pop_protocol/pop_protocol.erl [hunk 2]
```diff
@@ -30,6 +25,28 @@ compute_transaction_hash(Transaction) when is_map(Transaction) ->
 	 },
     my_crypto:hash( my_serializer:serialize_object(CleanTransaction) ).
 
+check_transaction_correctness(Transaction, ChainId) when is_map(Transaction) ->
+    #{
+      consensus_data := CD,
+      player_id := PlrKey
+     } = Transaction,
```
Comment (pdlla) [inline erlang/pop_protocol/pop_protocol.erl]: Does this crash if keys don't exist or if Transaction isn't a map (ignoring the fact there is is_map(...))

File: erlang/pop_protocol/pop_protocol.erl [hunk 3]
```diff
@@ -98,10 +109,14 @@ add_one_block(ProtocolData, Block, CurrentTime)
...
+    %% Verify transactions
+    lists:map(fun(T) -> check_transaction_correctness(T, MainChainId) end, BlockTransactionsList),
+
+    %% add this block to tree_data 
+    %% this can trigger errors if the block if poorly formed
+    %% also fails if block is orphan or already exists
+    TD1 = blocktree:add_new_block(Block, TD0),
 
-    pass.
+    TD1.
```
Comment (pdlla) [inline erlang/pop_protocol/pop_protocol.erl]: what does TD0/TD1 stand for?

File: erlang/potato_records.hrl [hunk 4]
```diff
@@ -0,0 +1,7 @@
+%% -record(transaction, {game_data, nonce, player_id, consensus_data}).
+%% -record(block, {previous_id, this_id, height, transactions, consensus_data}).
+-record(tree_data, {transaction_map=maps:new(), block_map=maps:new()}).
```
Comment (pdlla) [inline erlang/potato_records.hrl]: Nice!

though I do miss the good ole everything in tower.h/cpp days.

File: erlang/pop_protocol/pop_protocol.erl (hunk 2 above)
Comment (antonbobkov) [reply erlang/pop_protocol/pop_protocol.erl]: YA :O 
(yes for both)

File: erlang/compile_script (hunk 1 above)
Comment (antonbobkov) [reply erlang/compile_script]: -not -path './p2p/*' cause your stuff doesn't compile ಠ_ಠ

File: erlang/pop_protocol/pop_protocol.erl (hunk 3 above)
Comment (antonbobkov) [reply erlang/pop_protocol/pop_protocol.erl]: TD0 is initial state
TD1 is the modified final state 
(function modifies state by adding new block)

Comment (pdlla) [reply erlang/pop_protocol/pop_protocol.erl]: what does TD stand for? Tower Defense :o?

File: erlang/pop_protocol/pop_protocol.erl (hunk 2 above)
Comment (pdlla) [reply erlang/pop_protocol/pop_protocol.erl]: D:

File: erlang/compile_script (hunk 1 above)
Comment (pdlla) [reply erlang/compile_script]: :o

Comment (pdlla) [reply erlang/compile_script]: you're the one who code reviewed it :o

File: erlang/pop_protocol/pop_protocol.erl (hunk 3 above)
Comment (antonbobkov) [reply erlang/pop_protocol/pop_protocol.erl]: Tree Data lol
Actually it's incorrect! Should update the whole state not just tree data

Comment (antonbobkov) [reply erlang/pop_protocol/pop_protocol.erl]: Such code review O:

File: erlang/compile_script (hunk 1 above)
Comment (antonbobkov) [reply erlang/compile_script]: O:

**Pull request merged**